### DISCUSSION TITLE
Welcome to nuclear Discussions!

### DISCUSSION THREAD

**Discussion by nukeop** [nukeop/nuclear#806] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by FrostyPinkPlume** [comment]
Hello dear nuclear community.
I'm following you and using nuclear since long ago, but it's my first interaction since I can't compete with how competent you are.

In my family, we use neclear, but I had an idea. 
Electron is actually just a chromium browser connecting to a local web server. But what if I setup a webserver at home and use browsers to have Nuclear. If so, it would allow me to have only one instance to maintain and update.

That's what I wanted to do but :
- I'm unable to use Nuclear on the host.
- I'm unable to connect to the web server with another computer... (I don't even have the Nuclear logo)

Maybe this would be done by modifying the "main.js" file but I'm not able to locate it, since there are so many of them.

Thank you for your hard work.

**Reply by nukeop** [reply]
I'm not sure what you mean. You got it right that Electron is essentially a stripped-down Chromium, and that it opens a locally served page. Do you mean that you would like Nuclear to also be available via a regular browser? That would be possible, but we would need to remove some functionality in that version, since browsers don't offer some APIs that are available in Electron (such as access to the filesystem). In essence, you would need to heavily modify the web app (in `packages/app`) to detect whether it's running in Electron or in a browser, and make its features available depending on the detected environment.

**Reply by FrostyPinkPlume** [reply]
This would need a fork specially design for it... A "non electron" one... hmmmmm. It can't hurt to try.

**Reply by nukeop** [reply]
It can be done within the existing app, it would just require a lot of effort. Some features would need to be reworked completely, i.e. downloads, which are currently handled by Electron and would need to happen within the browser.

**Reply by FrostyPinkPlume** [reply]
So it would require to modify the app for a "server-version", a NON wich stand for Nuclear Over Network ^^.
French joke aside.

**Reply by nukeop** [reply]
Yep, I also had an idea where you could run the Electron version on a computer and it would serve a web version to other computers on the same local network, sort of like it currently exposes the API. This could have access to Electron APIs on the host machine and could save settings and download tracks there. Could be protected with a password too.

**Reply by FrostyPinkPlume** [reply]
I was more thinking of just having run with node js and making it available for the network. Having it headless allow to run it using small NASs like synolgy's ones.

**Reply by Clooos** [reply]
This is a great idea, any news about a server version?
